rest/authorization-server/src/main/java/org/eclipse/sw360/rest/authserver/security/customheaderauth/Sw360CustomHeaderAuthenticationProvider.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright Siemens AG, 2019. Part of the SW360 Portal Project.
+ *
+ * SPDX-License-Identifier: EPL-1.0
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.sw360.rest.authserver.security.customheaderauth;
+
+import org.eclipse.sw360.datahandler.permissions.PermissionUtils;
+import org.eclipse.sw360.datahandler.thrift.users.User;
+import org.eclipse.sw360.rest.authserver.Sw360AuthorizationServer;
+import org.eclipse.sw360.rest.authserver.security.Sw360GrantedAuthority;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.authentication.AuthenticationProvider;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.oauth2.common.util.OAuth2Utils;
+import org.springframework.security.oauth2.provider.ClientDetails;
+import org.springframework.security.oauth2.provider.ClientDetailsService;
+import org.springframework.security.web.authentication.preauth.PreAuthenticatedAuthenticationToken;
+
+import javax.annotation.PostConstruct;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * This {@link AuthenticationProvider} is specialised on requests where the
+ * {@link Sw360CustomHeaderAuthenticationFilter} made sure that the user has
+ * already been authenticated by some external proxy that set some headers to
+ * let us know about the authentication.
+ *
+ * In addition it is special because it calculates the granted authorities for
+ * the user depending on the user's authorities given by his groups and the
+ * client's scopes. The result will be the intersection between these two lists.
+ */
+public class Sw360CustomHeaderAuthenticationProvider implements AuthenticationProvider {
+
+    private final Logger log = Logger.getLogger(this.getClass());
```

Comment:
Yeah, but this seems a bit outdated to me. With the pushed way you get one logger for the complete class hierarchy in case of inheritance. I think this is more convenient.